strictdoc/backend/sdoc_source_code/models/source_node.py

```diff
@@ -13,16 +13,24 @@
 from strictdoc.backend.sdoc_source_code.models.range_marker import (
     RangeMarker,
 )
+from strictdoc.backend.sdoc_source_code.models.source_location import ByteRange
 from strictdoc.core.project_config import SourceNodesEntry
 
 
-@dataclass
+@dataclass(eq=False)
```

Comment:
`eq=False` means that dictionaries will index by object identity. Copied SourceNode objects will appear as two different SourceNodes. I'm not sure this is always what we want. An alternative could be to implement `__eq__` and `__hash__` so that they target `byte_range`.